Unable to pass a boolean input to readOnly (@rjsf/bootstrap-4)

### Prerequisites
* [x]  I have read the [documentation](https://react-jsonschema-form.readthedocs.io/);
* [x]  In the case of a bug report, I understand that providing a [SSCCE](http://sscce.org/) example is tremendously useful to the maintainers.
* [x]  Ideally, I'm providing a [sample JSFiddle](https://jsfiddle.net/n1k0/f2y3fq7L/6/) or a [shared playground link](https://rjsf-team.github.io/react-jsonschema-form/) demonstrating the issue.

### Description

Cannot pass a boolean input to readOnly with de Bootstrap4 theme but it is possible with MaterialUI theme.

### Steps to Reproduce

#### Imports

``` typescript
import React from "react";
import { Theme as Bootstrap4Theme } from "@rjsf/bootstrap-4";
import {
  withTheme
} from "@rjsf/core";
```

#### Code

``` typescript
  const JsonSchemaForm = withTheme(Bootstrap4Theme);
  return (
    <div style={{ position: "relative" }}>
      <JsonSchemaForm
	schema={{
 	 "title": "Entité",
	  "type": "object",
	  "properties": {
	    "@id": {
	      "type": "string",
 	     "title":"ID"
  	  },
  	  "nom": {
   	   "type": "string",
 	     "title":"Nom"
  	  },
  	  "prenom": {
  	    "type": "string"
  	  },
  	  "email": {
  	    "type": "string"
  	  },
   	 "age": {
   	   "type": "number"
  	  },
  	  "sdf": {
 	     "type": "boolean"
 	   }
	  },
	  "required": ["@id", "nom", "prenom", "email", "age", "sdf"]
	}}
        uiSchema={ {
          nom: {"ui:readonly": true},
          sdf: {"ui:readonly": true}
        }
        }
      />
      </JsonSchemaForm>
  );
```

#### Expected behavior

Have the input read only as the MaterialUI theme
Screenshot (materialUI theme):
![image](https://user-images.githubusercontent.com/72933545/109641758-5d6ea880-7b52-11eb-937e-41e142e7fe91.png)

#### Actual behavior

Simple boolean input
Screenshot (bootstrap theme)
![image](https://user-images.githubusercontent.com/72933545/109641487-0cf74b00-7b52-11eb-913e-370032deee19.png)

### Version

"@rjsf/bootstrap-4": "^2.4.2"
"@rjsf/core": "^2.3.0"
"@rjsf/material-ui": "^2.3.0"

The issue seems to be quite simple: In the bootstrap-4 Checkbox Widget we define a property `readOnly` on the react-bootstrap Form.Check component:

https://github.com/rjsf-team/react-jsonschema-form/blob/master/packages/bootstrap-4/src/CheckboxWidget/CheckboxWidget.tsx#L40

... but there is no such property for that component according to:

https://react-bootstrap.github.io/components/forms/#forms-form-check

So the solution should be something like:

```ts
        <Form.Check
          id={id}
          label={desc}
          checked={typeof value === "undefined" ? false : value}
          required={required}
          disabled={disabled || readonly}
          autoFocus={autofocus}
          onChange={_onChange}
          type="checkbox"
          onBlur={_onBlur}
          onFocus={_onFocus}
        />
```

I would submit a PR but I cannot start a fresh install of the project on a cloud based dev env (gitpod): build fail with:

```
@rjsf/material-ui: (typescript) Error: /workspace/react-jsonschema-form/packages/material-ui/src/FieldTemplate/FieldTemplate.tsx(39,7): semantic error TS2322: Type 'import("/workspace/react-jsonschema-form/packages/material-ui/node_modules/@rjsf/core/node_modules/@types/json-schema/index").JSONSchema7' is not assignable to type 'import("/workspace/react-jsonschema-form/packages/material-ui/node_modules/@types/json-schema/index").JSONSchema7'.
@rjsf/material-ui:   Types of property 'enum' are incompatible.
@rjsf/material-ui:     Type 'import("/workspace/react-jsonschema-form/packages/material-ui/node_modules/@rjsf/core/node_modules/@types/json-schema/index").JSONSchema7Type[] | undefined' is not assignable to type 'import("/workspace/react-jsonschema-form/packages/material-ui/node_modules/@types/json-schema/index").JSONSchema7Type[] | undefined'.
@rjsf/material-ui:       Type 'import("/workspace/react-jsonschema-form/packages/material-ui/node_modules/@rjsf/core/node_modules/@types/json-schema/index").JSONSchema7Type[]' is not assignable to type 'import("/workspace/react-jsonschema-form/packages/material-ui/node_modules/@types/json-schema/index").JSONSchema7Type[]'.
@rjsf/material-ui:         Type 'import("/workspace/react-jsonschema-form/packages/material-ui/node_modules/@rjsf/core/node_modules/@types/json-schema/index").JSONSchema7Type' is not assignable to type 'import("/workspace/react-jsonschema-form/packages/material-ui/node_modules/@types/json-schema/index").JSONSchema7Type'.
@rjsf/material-ui:           Type 'object' is not assignable to type 'JSONSchema7Type'.
@rjsf/material-ui:             Type '{}' is missing the following properties from type 'JSONSchema7Array': length, pop, push, concat, and 28 more.
@rjsf/material-ui: Error: /workspace/react-jsonschema-form/packages/material-ui/src/FieldTemplate/FieldTemplate.tsx(39,7): semantic error TS2322: Type 'import("/workspace/react-jsonschema-form/packages/material-ui/node_modules/@rjsf/core/node_modules/@types/json-schema/index").JSONSchema7' is not assignable to type 'import("/workspace/react-jsonschema-form/packages/material-ui/node_modules/@types/json-schema/index").JSONSchema7'.
@rjsf/material-ui:   Types of property 'enum' are incompatible.
@rjsf/material-ui:     Type 'import("/workspace/react-jsonschema-form/packages/material-ui/node_modules/@rjsf/core/node_modules/@types/json-schema/index").JSONSchema7Type[] | undefined' is not assignable to type 'import("/workspace/react-jsonschema-form/packages/material-ui/node_modules/@types/json-schema/index").JSONSchema7Type[] | undefined'.
@rjsf/material-ui:       Type 'import("/workspace/react-jsonschema-form/packages/material-ui/node_modules/@rjsf/core/node_modules/@types/json-schema/index").JSONSchema7Type[]' is not assignable to type 'import("/workspace/react-jsonschema-form/packages/material-ui/node_modules/@types/json-schema/index").JSONSchema7Type[]'.
@rjsf/material-ui:         Type 'import("/workspace/react-jsonschema-form/packages/material-ui/node_modules/@rjsf/core/node_modules/@types/json-schema/index").JSONSchema7Type' is not assignable to type 'import("/workspace/react-jsonschema-form/packages/material-ui/node_modules/@types/json-schema/index").JSONSchema7Type'.
@rjsf/material-ui:           Type 'object' is not assignable to type 'JSONSchema7Type'.
@rjsf/material-ui:             Type '{}' is missing the following properties from type 'JSONSchema7Array': length, pop, push, concat, and 28 more.
@rjsf/material-ui:     at error (/workspace/react-jsonschema-form/packages/material-ui/node_modules/rollup/dist/shared/node-entry.js:5400:30)
@rjsf/material-ui:     at throwPluginError (/workspace/react-jsonschema-form/packages/material-ui/node_modules/rollup/dist/shared/node-entry.js:11878:12)
@rjsf/material-ui:     at Object.error (/workspace/react-jsonschema-form/packages/material-ui/node_modules/rollup/dist/shared/node-entry.js:12912:24)
@rjsf/material-ui:     at Object.error (/workspace/react-jsonschema-form/packages/material-ui/node_modules/rollup/dist/shared/node-entry.js:12081:38)
@rjsf/material-ui:     at RollupContext.error (/workspace/react-jsonschema-form/packages/material-ui/node_modules/rollup-plugin-typescript2/dist/rollup-plugin-typescript2.cjs.js:17194:30)
@rjsf/material-ui:     at /workspace/react-jsonschema-form/packages/material-ui/node_modules/rollup-plugin-typescript2/dist/rollup-plugin-typescript2.cjs.js:24994:23
@rjsf/material-ui:     at arrayEach (/workspace/react-jsonschema-form/packages/material-ui/node_modules/rollup-plugin-typescript2/dist/rollup-plugin-typescript2.cjs.js:534:11)
@rjsf/material-ui:     at forEach (/workspace/react-jsonschema-form/packages/material-ui/node_modules/rollup-plugin-typescript2/dist/rollup-plugin-typescript2.cjs.js:9360:14)
@rjsf/material-ui:     at printDiagnostics (/workspace/react-jsonschema-form/packages/material-ui/node_modules/rollup-plugin-typescript2/dist/rollup-plugin-typescript2.cjs.js:24967:5)
@rjsf/material-ui:     at Object.transform (/workspace/react-jsonschema-form/packages/material-ui/node_modules/rollup-plugin-typescript2/dist/rollu
lerna ERR! npm run build exited 1 in '@rjsf/material-ui'
```

Yeah, it's weird. I just used nvm to get node 14. I destroyed my local and re-cloned, then updated npm to latest. Re-installed, rebuilt and got the same error. I'm on a Mac, Big Sur.

```
@rjsf/material-ui: (typescript) Error: /Users/paintedbicycle/Sites/react-jsonschema-form/packages/material-ui/src/FieldTemplate/FieldTemplate.tsx(39,7): semantic error TS2322: Type 'import("/Users/paintedbicycle/Sites/react-jsonschema-form/packages/material-ui/node_modules/@rjsf/core/node_modules/@types/json-schema/index").JSONSchema7' is not assignable to type 'import("/Users/paintedbicycle/Sites/react-jsonschema-form/packages/material-ui/node_modules/@types/json-schema/index").JSONSchema7'.
@rjsf/material-ui:   Types of property 'enum' are incompatible.
@rjsf/material-ui:     Type 'import("/Users/paintedbicycle/Sites/react-jsonschema-form/packages/material-ui/node_modules/@rjsf/core/node_modules/@types/json-schema/index").JSONSchema7Type[] | undefined' is not assignable to type 'import("/Users/paintedbicycle/Sites/react-jsonschema-form/packages/material-ui/node_modules/@types/json-schema/index").JSONSchema7Type[] | undefined'.
@rjsf/material-ui:       Type 'import("/Users/paintedbicycle/Sites/react-jsonschema-form/packages/material-ui/node_modules/@rjsf/core/node_modules/@types/json-schema/index").JSONSchema7Type[]' is not assignable to type 'import("/Users/paintedbicycle/Sites/react-jsonschema-form/packages/material-ui/node_modules/@types/json-schema/index").JSONSchema7Type[]'.
@rjsf/material-ui:         Type 'import("/Users/paintedbicycle/Sites/react-jsonschema-form/packages/material-ui/node_modules/@rjsf/core/node_modules/@types/json-schema/index").JSONSchema7Type' is not assignable to type 'import("/Users/paintedbicycle/Sites/react-jsonschema-form/packages/material-ui/node_modules/@types/json-schema/index").JSONSchema7Type'.
@rjsf/material-ui:           Type 'object' is not assignable to type 'JSONSchema7Type'.
@rjsf/material-ui:             Type '{}' is missing the following properties from type 'JSONSchema7Array': length, pop, push, concat, and 28 more.
@rjsf/material-ui: Error: /Users/paintedbicycle/Sites/react-jsonschema-form/packages/material-ui/src/FieldTemplate/FieldTemplate.tsx(39,7): semantic error TS2322: Type 'import("/Users/paintedbicycle/Sites/react-jsonschema-form/packages/material-ui/node_modules/@rjsf/core/node_modules/@types/json-schema/index").JSONSchema7' is not assignable to type 'import("/Users/paintedbicycle/Sites/react-jsonschema-form/packages/material-ui/node_modules/@types/json-schema/index").JSONSchema7'.
@rjsf/material-ui:   Types of property 'enum' are incompatible.
@rjsf/material-ui:     Type 'import("/Users/paintedbicycle/Sites/react-jsonschema-form/packages/material-ui/node_modules/@rjsf/core/node_modules/@types/json-schema/index").JSONSchema7Type[] | undefined' is not assignable to type 'import("/Users/paintedbicycle/Sites/react-jsonschema-form/packages/material-ui/node_modules/@types/json-schema/index").JSONSchema7Type[] | undefined'.
@rjsf/material-ui:       Type 'import("/Users/paintedbicycle/Sites/react-jsonschema-form/packages/material-ui/node_modules/@rjsf/core/node_modules/@types/json-schema/index").JSONSchema7Type[]' is not assignable to type 'import("/Users/paintedbicycle/Sites/react-jsonschema-form/packages/material-ui/node_modules/@types/json-schema/index").JSONSchema7Type[]'.
@rjsf/material-ui:         Type 'import("/Users/paintedbicycle/Sites/react-jsonschema-form/packages/material-ui/node_modules/@rjsf/core/node_modules/@types/json-schema/index").JSONSchema7Type' is not assignable to type 'import("/Users/paintedbicycle/Sites/react-jsonschema-form/packages/material-ui/node_modules/@types/json-schema/index").JSONSchema7Type'.
@rjsf/material-ui:           Type 'object' is not assignable to type 'JSONSchema7Type'.
@rjsf/material-ui:             Type '{}' is missing the following properties from type 'JSONSchema7Array': length, pop, push, concat, and 28 more.
@rjsf/material-ui:     at error (/Users/paintedbicycle/Sites/react-jsonschema-form/packages/material-ui/node_modules/rollup/dist/shared/node-entry.js:5400:30)
@rjsf/material-ui:     at throwPluginError (/Users/paintedbicycle/Sites/react-jsonschema-form/packages/material-ui/node_modules/rollup/dist/shared/node-entry.js:11878:12)
@rjsf/material-ui:     at Object.error (/Users/paintedbicycle/Sites/react-jsonschema-form/packages/material-ui/node_modules/rollup/dist/shared/node-entry.js:12912:24)
@rjsf/material-ui:     at Object.error (/Users/paintedbicycle/Sites/react-jsonschema-form/packages/material-ui/node_modules/rollup/dist/shared/node-entry.js:12081:38)
@rjsf/material-ui:     at RollupContext.error (/Users/paintedbicycle/Sites/react-jsonschema-form/packages/material-ui/node_modules/rollup-plugin-typescript2/dist/rollup-plugin-typescript2.cjs.js:17194:30)
@rjsf/material-ui:     at /Users/paintedbicycle/Sites/react-jsonschema-form/packages/material-ui/node_modules/rollup-plugin-typescript2/dist/rollup-plugin-typescript2.cjs.js:24994:23
@rjsf/material-ui:     at arrayEach (/Users/paintedbicycle/Sites/react-jsonschema-form/packages/material-ui/node_modules/rollup-plugin-typescript2/dist/rollup-plugin-typescript2.cjs.js:534:11)
@rjsf/material-ui:     at forEach (/Users/paintedbicycle/Sites/react-jsonschema-form/packages/material-ui/node_modules/rollup-plugin-typescript2/dist/rollup-plugin-typescript2.cjs.js:9360:14)
@rjsf/material-ui:     at printDiagnostics (/Users/paintedbicycle/Sites/react-jsonschema-form/packages/material-ui/node_modules/rollup-plugin-typescript2/dist/rollup-plugin-typescript2.cjs.js:24967:5)
@rjsf/material-ui:     at Object.transform (/Users/paintedbicycle/Sites/react-jsonschema-form/packages/material-ui/node_modules/rollup-plugin-typescript2/dist/rollup-plugin-typescript2.cjs.js:29049:17)
@rjsf/material-ui: npm ERR! code 1
@rjsf/material-ui: npm ERR! path /Users/paintedbicycle/Sites/react-jsonschema-form/packages/material-ui
@rjsf/material-ui: npm ERR! command failed
@rjsf/material-ui: npm ERR! command sh -c rimraf dist && tsdx build
@rjsf/material-ui: npm ERR! A complete log of this run can be found in:
```

Hmmm... looking at the error you're getting:

```
@rjsf/material-ui: (typescript) Error: /Users/paintedbicycle/Sites/react-jsonschema-form/packages/material-ui/src/FieldTemplate/FieldTemplate.tsx(39,7): semantic error TS2322: Type 'import("/Users/paintedbicycle/Sites/react-jsonschema-form/packages/material-ui/node_modules/@rjsf/core/node_modules/@types/json-schema/index").JSONSchema7' is not assignable to type 'import("/Users/paintedbicycle/Sites/react-jsonschema-form/packages/material-ui/node_modules/@types/json-schema/index").JSONSchema7'.
```

Perhaps you have two different versions of `@types/json-schema` installed? Can you check the versions of `@types/json-schema` in `packages/material-ui/package-lock.json` and in `packages/core/package-lock.json`? They should match -- otherwise we might need to fix the package-lock.json files.

A better solution for the long term, though, would be to share dependencies using either `lerna bootstrap --hoist` or yarn workspaces.

--- a/packages/bootstrap-4/src/CheckboxWidget/CheckboxWidget.tsx
+++ b/packages/bootstrap-4/src/CheckboxWidget/CheckboxWidget.tsx
@@ -36,8 +36,7 @@ const CheckboxWidget = (props: WidgetProps) => {
           label={desc}
           checked={typeof value === "undefined" ? false : value}
           required={required}
-          disabled={disabled}
-          readOnly={readonly}
+          disabled={disabled || readonly}
           autoFocus={autofocus}
           onChange={_onChange}
           type="checkbox"

--- a/packages/bootstrap-4/src/CheckboxesWidget/CheckboxesWidget.tsx
+++ b/packages/bootstrap-4/src/CheckboxesWidget/CheckboxesWidget.tsx
@@ -73,8 +73,7 @@ const CheckboxesWidget = ({
                 onChange={_onChange(option)}
                 onBlur={_onBlur}
                 onFocus={_onFocus}
-                disabled={disabled || itemDisabled || false}
-                readOnly={readonly}
+                disabled={disabled || readonly || itemDisabled || false}
               />
             </Form>
           ) : (
@@ -91,8 +90,7 @@ const CheckboxesWidget = ({
                 onChange={_onChange(option)}
                 onBlur={_onBlur}
                 onFocus={_onFocus}
-                disabled={disabled || itemDisabled || false}
-                readOnly={readonly}
+                disabled={disabled || readonly || itemDisabled || false}
               />
             </Form>
           );

--- a/packages/bootstrap-4/src/CheckboxesWidget/CheckboxesWidget.tsx
+++ b/packages/bootstrap-4/src/CheckboxesWidget/CheckboxesWidget.tsx
@@ -73,7 +73,7 @@ const CheckboxesWidget = ({
                 onChange={_onChange(option)}
                 onBlur={_onBlur}
                 onFocus={_onFocus}
-                disabled={disabled || readonly || itemDisabled || false}
+                disabled={disabled || itemDisabled || readonly}
               />
             </Form>
           ) : (

--- a/packages/bootstrap-4/src/CheckboxesWidget/CheckboxesWidget.tsx
+++ b/packages/bootstrap-4/src/CheckboxesWidget/CheckboxesWidget.tsx
@@ -90,7 +90,7 @@ const CheckboxesWidget = ({
                 onChange={_onChange(option)}
                 onBlur={_onBlur}
                 onFocus={_onFocus}
-                disabled={disabled || readonly || itemDisabled || false}
+                disabled={disabled || itemDisabled || readonly}
               />
             </Form>
           );

